cmake/create_test.cmake

```diff
@@ -167,55 +167,14 @@ endfunction(ginkgo_create_cuda_test_internal)
     ginkgo_build_test_name(${test_name} test_target_name)
-    ginkgo_create_hip_test_internal(${test_name} ${test_name}.hip.cpp ${test_target_name} "" ${ARGN})
+    ginkgo_create_hip_test_internal(${test_name} ${test_name}.hip.cpp ${test_target_name} ${ARGN})
 endfunction(ginkgo_create_hip_test)
 
 ## Internal function allowing separate filename, test name and test target name.
-function(ginkgo_create_hip_test_internal test_name filename test_target_name additional_flags)
-    set_source_files_properties(${filename} PROPERTIES HIP_SOURCE_PROPERTY_FORMAT TRUE)
-    set(GINKGO_TEST_HIP_DEFINES -DGKO_COMPILING_HIP ${additional_flags})
-    if (GINKGO_FAST_TESTS)
-        list(APPEND GINKGO_TEST_HIP_DEFINES -DGINKGO_FAST_TESTS)
-    endif()
-    if (GINKGO_TEST_NONDEFAULT_STREAM)
-        list(APPEND GINKGO_TEST_HIP_DEFINES -DGKO_TEST_NONDEFAULT_STREAM)
-    endif()
-
-    # NOTE: With how HIP works, passing the flags `HIPCC_OPTIONS` etc. here
-    # creates a redefinition of all flags. This creates some issues with `nvcc`,
-    # but `clang` seems fine with the redefinitions.
-    if (GINKGO_HIP_PLATFORM MATCHES "${HIP_PLATFORM_NVIDIA_REGEX}")
-        hip_add_executable(${test_target_name} ${filename}
-            # If `FindHIP.cmake`, namely `HIP_PARSE_HIPCC_OPTIONS` macro and
-            # call gets fixed, uncomment this.
-            HIPCC_OPTIONS ${GINKGO_TEST_HIP_DEFINES} # ${GINKGO_HIPCC_OPTIONS}
-            # NVCC_OPTIONS  ${GINKGO_TEST_HIP_DEFINES} ${GINKGO_HIP_NVCC_OPTIONS}
-            # CLANG_OPTIONS ${GINKGO_TEST_HIP_DEFINES} ${GINKGO_HIP_CLANG_OPTIONS}
-            --expt-relaxed-constexpr --expt-extended-lambda
-            )
-    else() # hcc/clang
-        hip_add_executable(${test_target_name} ${filename}
-            HIPCC_OPTIONS ${GINKGO_HIPCC_OPTIONS} ${GINKGO_TEST_HIP_DEFINES}
-            NVCC_OPTIONS  ${GINKGO_HIP_NVCC_OPTIONS}
-            CLANG_OPTIONS ${GINKGO_HIP_CLANG_OPTIONS}
-            )
-    endif()
-
-    # Let's use a normal compiler for linking
-    set_target_properties(${test_target_name} PROPERTIES LINKER_LANGUAGE CXX)
-
-    target_include_directories(${test_target_name}
-        PRIVATE
-        # Only `math` requires it so far, but it's much easier
-        # to put these this way.
-        ${GINKGO_HIP_THRUST_PATH}
-        # Only `exception_helpers` requires these so far, but it's much easier
-        # to put these this way.
-        ${HIPBLAS_INCLUDE_DIRS}
-        ${HIPFFT_INCLUDE_DIRS}
-        ${hiprand_INCLUDE_DIRS}
-        ${HIPSPARSE_INCLUDE_DIRS}
-        )
+function(ginkgo_create_hip_test_internal test_name filename test_target_name)
+    set_source_files_properties(${filename} PROPERTIES LANGUAGE HIP)
+    add_executable(${test_target_name} ${filename})
+    target_compile_definitions(${test_target_name} PRIVATE GKO_COMPILING_HIP)
     ginkgo_set_test_target_properties(${test_target_name} "_hip" ${ARGN})
     ginkgo_add_test(${test_name} ${test_target_name} ${ARGN} RESOURCE_TYPE hipgpu)
```

Comment:
I assume the cxx standard will simply propagate here when we already set ginkgo_hip outside.
If not, maybe also set the standard here

Reply:
good point, the proper way to do it would be setting `CUDA_STANDARD` and `HIP_STANDARD`, which we neglect to do. It works right now, because the compilers default to C++17